torch_uncertainty/datasets/classification/ucr_uea.py

```diff
@@ -0,0 +1,143 @@
+            raise ValueError(
+                "Cannot use split 'ood' without setting create_ood to True. "
+                "Set create_ood=True to create an OOD dataset."
+            )
+
+        super().__init__()
+        self.dataset_name = dataset_name
+        self.split = split
+        self.transform = transform
+        self.target_transform = target_transform
+        self.create_ood = create_ood
+
+        if create_ood:
+            warnings.warn(
+                "When `create_ood` is set to True, the dataset will be created with the last class as OOD.",
+                UserWarning,
+                stacklevel=2,
+            )
+
+        self._make_dataset()
+
+    def __len__(self) -> int:
+        """Get the number of samples in the dataset.
+
+        Returns:
+            int: The number of samples in the dataset.
+        """
+        return len(self.data[self.split][0])
+
+    def __getitem__(self, index: int) -> tuple[torch.Tensor, torch.Tensor]:
+        """Get a data sample from the dataset.
+
+        Args:
+            index (int): The index of the sample to retrieve.
+
+        Returns:
+            tuple[torch.Tensor, torch.Tensor]: A tuple containing the input tensor and the target tensor.
+        """
+        inputs, target = self.data[self.split][0][index], self.data[self.split][1][index]
+        inputs = rearrange(inputs, "t c -> c t")  # Change shape to (channels, time)
+
+        if self.transform:
+            inputs = self.transform(inputs)
+
+        if self.target_transform:
+            target = self.target_transform(target)
+
+        return inputs, target
+
+    def _make_dataset(self):
+        """Create the dataset from the loaded data."""
+        dataloader = UCR_UEA_datasets()
+        x_train, y_train, x_test, y_test = dataloader.load_dataset(self.dataset_name)
+
+        self.classes = np.unique(y_train)
+        self.min_label = np.min(self.classes)
+
+        y_train = y_train - self.min_label  # Normalize labels to start from 0
+        y_test = y_test - self.min_label  # Normalize labels to start from 0
+
+        ood_data = None
+
+        if self.create_ood:
+            train_ood_mask = y_train == (len(self.classes) - 1)
+            test_ood_mask = y_test == (len(self.classes) - 1)
+
+            x_train_ood = x_train[train_ood_mask]
+            y_train_ood = y_train[train_ood_mask]
+            x_test_ood = x_test[test_ood_mask]
+            y_test_ood = y_test[test_ood_mask]
+            x_ood = np.concatenate((x_train_ood, x_test_ood), axis=0)
+            y_ood = np.concatenate((y_train_ood, y_test_ood), axis=0)
+
+            x_train = x_train[~train_ood_mask]
+            y_train = y_train[~train_ood_mask]
+            x_test = x_test[~test_ood_mask]
+            y_test = y_test[~test_ood_mask]
+
+            ood_data = [
+                torch.tensor(x_ood, dtype=torch.float32),
+                torch.tensor(y_ood, dtype=torch.int64),
```

Comment:
use torch.long instead of torch.int64
```suggestion
                torch.tensor(y_ood, dtype=torch.long),
```